backend/src/main/java/pl/wroclaw/asi/labdaybackendspring/controllers/AppDataController.java

```diff
@@ -0,0 +1,91 @@
+package pl.wroclaw.asi.labdaybackendspring.controllers;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import pl.wroclaw.asi.labdaybackendspring.model.AppData;
+import pl.wroclaw.asi.labdaybackendspring.model.LastUpdate;
+import pl.wroclaw.asi.labdaybackendspring.services.*;
+
+import java.security.Principal;
+import java.sql.Timestamp;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/api")
+public class AppDataController {
+
+    private final EventService eventService;
+    private final PlaceService placeService;
+    private final PathService pathService;
+    private final TimetableService timetableService;
+    private final SpeakerService speakerService;
+    private final LastUpdateService lastUpdateService;
+
+    private final ValidationErrorService validationErrorService;
+
+    public AppDataController(
```

Comment:
This is wrong, injecting more than 3 services is a sign of wrong architecture. Is it necessary?

Reply:
I agree, but I don't know how to handle it due to we need to return Appdata which is composed of five models.

Reply:
The problem is apparently in `AppData.java` design. Passing 5 list objects in constructor is a bad move and you know it. That's why you have to have 7 fields in constructor in `AppDataController.java`. Do you need all the data available at once? Does the mobile app requires it?

Reply:
refactored here: https://github.com/asi-pwr/LabDayBackendSpring/pull/24